dissect/target/plugins/os/unix/linux/proc.py

```diff
@@ -53,71 +53,68 @@ class NetSocket:
-        self.local_port, self.remote_port = [
+        # So we use the fields(NetSocket) to dynamically fill the NetSocket dataclass.
+        for idx, value in enumerate(line.split()):
+            field_names = fields(cls)
+            setattr(socket, field_names[idx].name, value)
+
+        socket.uid = int(socket.uid)
+        socket.inode = int(socket.inode)
+        socket.tx_queue, socket.rx_queue = [int(queue, 16) for queue in socket.tx_rx_queue.split(":")]
+        socket.local_port, socket.remote_port = [
             int(port, 16)
             for port in (
-                self.local_address.split(":")[1],
-                self.rem_address.split(":")[1],
+                socket.local_address.split(":")[1],
+                socket.rem_address.split(":")[1],
             )
         ]
 
-        return self
+        return socket
 
 
 @dataclass(order=True, init=False)
 class UnixSocket:
-    num: str  # the kernel table slot number.
-    ref: int  # the number of users of the socket.
-    protocol: int  # currently always 0. "Unix"
-    flags: str  # the internal kernel flags holding the status of the socket.
-    type: int  # the socket type: 1 for SOCK_STREAM, 2 for SOCK_DGRAM and 5 for SOCK_SEQPACKET sockets
-    state: int  # the internal state of the socket.
-    inode: int  # the inode number of the socket. the inode is commonly refered to as port in tools as ss and netstat
+    num: str = field(default=None)  # the kernel table slot number.
+    ref: int = field(default=0)  # the number of users of the socket.
+    protocol: int = field(default=0)  # currently always 0. "Unix"
+    flags: str = field(default=None)  # the internal kernel flags holding the status of the socket.
+    type: int = field(
+        default=0
+    )  # the socket type: 1 for SOCK_STREAM, 2 for SOCK_DGRAM and 5 for SOCK_SEQPACKET sockets
+    state: int = field(default=0)  # the internal state of the socket.
+    inode: int = field(
+        default=0
+    )  # the inode number of the socket. the inode is commonly refered to as port in tools as ss and netstat
     path: str = field(default=None)  # sockets in the abstract namespace are included in the list,
     # and are shown with a Path that commences with the character '@'.
 
     # Values parsed from raw values listed above.
     state_string: str = field(default=None)
     stream_type_string: str = field(default=None)
-    protocol_string: str = field(default=None)
-
-    # Same as defining field_name: type = field(init=False, default=None)
-    def __init__(self):
-        for fld in fields(self):
-            setattr(self, fld.name, None)
+    protocol_string: str = field(default="unix")
 
-        self.protocol_string = "unix"
+    @classmethod
+    def from_line(cls, line: str) -> UnixSocket:
+        socket = cls()
 
-    def from_line(self, line: str) -> UnixSocket:
         for idx, value in enumerate(line.split()):
```

Comment:
Would `socket = cls(*line.split())` also work? Same for the other places where you do something like this btw.